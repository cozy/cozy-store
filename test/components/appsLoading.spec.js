'use strict'

/* eslint-env jest */

import { render } from '@testing-library/react'
import {
  AppsLoading,
  LoadingAppsComponents
} from 'ducks/components/AppsLoading'
import React from 'react'

global.Math.random = () => 1 // remove random for testing

describe('AppsLoading component', () => {
  it('should be rendered correctly and never be updated', () => {
    const spy = jest.spyOn(AppsLoading.prototype, 'shouldComponentUpdate')
    render(<AppsLoading />)
    expect(spy).not.toHaveBeenCalled()
    spy.mockRestore()
  })
})

describe('LoadingAppsComponents component', () => {
  it('should be rendered correctly if isMobile', () => {
    const { asFragment } = render(
      <LoadingAppsComponents
        count={3}
        subKey="mock"
        breakpoints={{ isMobile: true }}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
