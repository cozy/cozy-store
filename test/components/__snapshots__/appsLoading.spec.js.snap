// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingAppsComponents component should be rendered correctly if isMobile 1`] = `
<DocumentFragment>
  <div
    class="sto-sections-list"
  >
    <div
      class="sto-small-app-item"
    >
      <div
        class="sto-small-app-item-icon-wrapper"
      >
        <div
          class="sto-small-app-item-icon"
        >
          <span
            class="sto-sections-placeholder"
            data-testid="placeholder"
            style="width: 2.5rem; height: 2.5rem; margin: auto;"
          />
        </div>
      </div>
      <div
        class="sto-small-app-item-desc"
      >
        <h4
          class="sto-small-app-item-title"
        >
          <span
            class="sto-sections-placeholder"
            data-testid="placeholder"
            style="width: 5rem; height: .8rem; margin: auto;"
          />
        </h4>
        <p
          class="sto-small-app-item-developer"
        >
          <span
            class="sto-sections-placeholder"
            data-testid="placeholder"
            style="width: 5rem; height: .8rem; margin: auto;"
          />
        </p>
        <p
          class="sto-small-app-item-status"
        >
          <span
            class="sto-sections-placeholder"
            data-testid="placeholder"
            style="width: 4rem; height: .6rem; margin: auto;"
          />
        </p>
      </div>
    </div>
    <div
      class="sto-small-app-item"
    >
      <div
        class="sto-small-app-item-icon-wrapper"
      >
        <div
          class="sto-small-app-item-icon"
        >
          <span
            class="sto-sections-placeholder"
            data-testid="placeholder"
            style="width: 2.5rem; height: 2.5rem; margin: auto;"
          />
        </div>
      </div>
      <div
        class="sto-small-app-item-desc"
      >
        <h4
          class="sto-small-app-item-title"
        >
          <span
            class="sto-sections-placeholder"
            data-testid="placeholder"
            style="width: 5rem; height: .8rem; margin: auto;"
          />
        </h4>
        <p
          class="sto-small-app-item-developer"
        >
          <span
            class="sto-sections-placeholder"
            data-testid="placeholder"
            style="width: 5rem; height: .8rem; margin: auto;"
          />
        </p>
        <p
          class="sto-small-app-item-status"
        >
          <span
            class="sto-sections-placeholder"
            data-testid="placeholder"
            style="width: 4rem; height: .6rem; margin: auto;"
          />
        </p>
      </div>
    </div>
    <div
      class="sto-small-app-item"
    >
      <div
        class="sto-small-app-item-icon-wrapper"
      >
        <div
          class="sto-small-app-item-icon"
        >
          <span
            class="sto-sections-placeholder"
            data-testid="placeholder"
            style="width: 2.5rem; height: 2.5rem; margin: auto;"
          />
        </div>
      </div>
      <div
        class="sto-small-app-item-desc"
      >
        <h4
          class="sto-small-app-item-title"
        >
          <span
            class="sto-sections-placeholder"
            data-testid="placeholder"
            style="width: 5rem; height: .8rem; margin: auto;"
          />
        </h4>
        <p
          class="sto-small-app-item-developer"
        >
          <span
            class="sto-sections-placeholder"
            data-testid="placeholder"
            style="width: 5rem; height: .8rem; margin: auto;"
          />
        </p>
        <p
          class="sto-small-app-item-status"
        >
          <span
            class="sto-sections-placeholder"
            data-testid="placeholder"
            style="width: 4rem; height: .6rem; margin: auto;"
          />
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
